src/dmd/lambdacomp.d

```diff
@@ -17,51 +29,81 @@ import dmd.statement;
 import dmd.tokens;
 import dmd.visitor;
 
+/**
+ * This enum is used to report the type of the expression
+ * found when visiting the AST nodes to higher levels of the
+ * AST.
```

Comment:
I don't know what `visiting the AST nodes to the higher levels of the AST` means.